src/server/maps/MapleReactor.java

```diff
@@ -261,10 +261,11 @@ public synchronized void hitReactor(boolean wHit, int charPos, short stance, int
                                     map.broadcastMessage(MaplePacketCreator.triggerReactor(this, stance));
                                 }
 
-                                ReactorScriptManager.getInstance().act(c, this);
+                                ReactorScriptManager.getInstance().act(c, this);                                
                             } else { //reactor not broken yet
                                 map.broadcastMessage(MaplePacketCreator.triggerReactor(this, stance));
-                                if (state == stats.getNextState(state, b)) {//current state = next state, looping reactor
+                                if (state == stats.getNextState(state, b) || //current state = next state, looping reactor
+                                        (state == 4 && stats.getType(state) == 101)) { //next state = 0 is also a looping reactor
```

Comment:
can you explain the conditional `(state == 4 && stats.getType(state) == 101)`

Reply:
First dungeon in Neo City (240070100 - 240070102) uses a special reactor that restores itself instead of being destroyed. State 4 is the state at which the reactor enters a timeout, during which it does not respond to hits, and after which it will enter the next state as defined in the WZ file. Type 101 appears to indicate that the reactor is looping.